### ISSUE TITLE
Possible bug in ValleyRackFree/Plateau LFO modulation?

### ISSUE THREAD

**Issue by WeirdConstructor** [ValleyAudio/ValleyRackFree#61]
Hi, I've studied the source code of the Dattorro reverb and stumbled across these lines of the all-pass modulation:

```cpp
    // In Dattorro::process():
    _leftApf1.delay.setDelayTime(_lfo1.process() * _lfoDepth * modDepth + _leftApf1Time);
    _leftApf2.delay.setDelayTime(_lfo2.process() * _lfoDepth * modDepth + _leftApf2Time);
    _rightApf1.delay.setDelayTime(_lfo3.process() * _lfoDepth * modDepth + _rightApf1Time);
    _rightApf2.delay.setDelayTime(_lfo4.process() * _lfoDepth * modDepth + _rightApf2Time);
```
If I am not mistaken, `_lfo1` is a TriSawLFO, which does return a value range of -1 to 1:
```cpp
    TriSawLFO::process() {
        // ...
        _output *= 2.0;
        _output -= 1.0;
        return _output;
    }
```
The `setDelayTime` handles negative values by clamping them to `0`.
But doesn't that mean, that half of the modulation time the all-pass filters are not doing anything?
Maybe I am missing something.

**Comment by WeirdConstructor** [comment]
As a second thing I found: It also seems, that the `modDepth` has a rather weird value. Because the input voltage `0.0` to `10.0` is mapped to `0.0` and `16.0`. 16 is a rather odd value, and it's in fact the Excursion.

```cpp
// In Dattorro.cpp/.hpp:
    const double _kLfoExcursion = 16.0;
    // ...
    _lfoDepth = dattorroScale(_kLfoExcursion);

// In Plateaeu.cpp/.hpp
    const float modDepthMin = 0.f;
    const float modDepthMax = 16.f;
    // ...
    modDepth = inputs[MOD_DEPTH_CV_INPUT].getVoltage() * params[MOD_DEPTH_CV_PARAM].getValue();
    modDepth = rescale(modDepth, 0.f, 10.f, modDepthMin, modDepthMax);
    modDepth += params[MOD_DEPTH_PARAM].getValue();
    modDepth = clamp(modDepth, modDepthMin, modDepthMax);
    // ...
    reverb.modDepth = modDepth;
```

This means at medium or high modDepth values the excursion is way exceeding what was originally intended by Dattorro. The parameter does indeed lead to rather weird sounds at these settings. I guess it does not matter much after all :-)

**Comment by WeirdConstructor** [comment]
Ah never mind, I think these don't matter after all :-)

**Comment by ValleyAudio** [comment]
Hi @WeirdConstructor , apologies for the delayed reply. I’ve actually been distant from Rack development for a while. Thanks for looking over my code though, it’s good to hear other opinions and suggestions.

Anyway, a couple of months go I overhauled Plateau from the ground up, however I have not merged this personal code to my public v1.0 branch until I am satisfied that the module sounds the same, I don’t want to upset people with any possible breaking changes to this module you see. The overhaul I performed pretty much addresses the issues you have raised amongst other features such as block processing to reduce CPU usage. As for negative delay times caused by the bipolar LFO output, it only causes issues at very short “Size” value parameter. They’re clamped to 0 so you don’t hear glitches from the delay line wrapping back on it’s own circular buffer.

I hope to push this overhaul soon, but thanks for taking the time look at my code and comment :)

**Comment by WeirdConstructor** [comment]
Hi @ValleyAudio!

Thanks a lot for the reply.
Yea, I've seen that even with a negative modulation it does by far not exceed `_leftApf1Time`, so I interpreted that wrongly and it's absolutely no problem after all. The `16 * 16` LFO excursion was indeed weird, but you could count that as "interesting" effect - especially with the directed ramps of the TriSawLFO!

I hope with the overhaul you also took care of this?
```cpp
    const long _kLeftTaps[7] = {266, 2974, 1913, 1996, 1990, 187, 1066};
    const long _kRightTaps[7] = {266, 2974, 1913, 1996, 1990, 187, 1066};
```
Dattorro had different taps for the right line. The effect is not that big after all.

The reason I looked so closely at your code was, that you had a very readable and clean implementation and I needed a good reverb for ambient/atmospheric/spacey music. And Plateau still seems to be the best open source reverb out there for VCVRack. So thank you a lot for your work! I've reimplemented this now in Rust for my own modular synthesizer project ([HexoSynth](https://github.com/WeirdConstructor/HexoSynth), not VCV Rack related). In case you are curious, here is my code: https://github.com/WeirdConstructor/HexoDSP/blob/master/src/dsp/dattorro.rs
